ComponentKit/Core/Scope/CKDetectComponentScopeCollisions.mm

```diff
@@ -31,9 +31,53 @@
 }
 #pragma clang diagnostic pop
 
-void CKDetectComponentScopeCollisions(const CKComponentLayout &layout)
+static CKComponent *completeComponentScopeCollisionPair(CKComponent *collidingComponent,
+                                            id<NSObject> collidingScope,
+                                            NSMapTable<CKComponent *, CKComponent*> *componentsToParentComponents)
+{
+  for (CKComponent* componentKey in componentsToParentComponents) {
+    const id<NSObject> scopeFrameToken = [componentKey scopeFrameToken];
+    if ([collidingScope isEqual:scopeFrameToken] && ![collidingComponent isEqual:componentKey]) {
+      return componentKey;
+    }
+  }
+  return nil;
+}
+
+static CKComponent *lowestCommonAncestor(CKComponent *component,
+                                         id<NSObject> collisionScope,
+                                         NSMapTable<CKComponent *, CKComponent*> *componentsToParentComponents)
+{
+  // First we need to find the node we're having a collision with.
+  CKComponent *collidingComponent = completeComponentScopeCollisionPair(component, collisionScope, componentsToParentComponents);
+  if ([collidingComponent isEqual:component]) {
+    return [componentsToParentComponents objectForKey:component];
```

Comment:
I don't think that situation would ever arise. That the colliding component is the same as the component. But I'm not sure of the internals of componentsKit, so I tried to play safe. However It's not needed anyway. If that situation happens, the rest of the code would take care anyway. Thanks for pointing it out.